File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {



        return mysol(s);
    }
    bool mysol(string s){
        unordered_map<char, char> mapp = { 
    {'}', '{'},
    {')', '('}, 
    {']', '['}
};
        stack<char> st;

        for(int i=0;i<s.size();i++){
            char ch = s[i];
            if(mapp.find(ch)!=mapp.end()){
                // Its a closing brack
                if(st.empty() || st.top() != mapp[ch]) return false;
                st.pop();
            }
            else{
                st.push(ch);
            }    
        }
        return st.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        list_1= []
        map = { 
            '{':'}', 
            '(':')',
            '[':']'
        }
        for x in s:
            # if x in (')','}',']'):
            #     return False
            if x in map.keys():
                list_1.append(map[x])
                # print(list_1)
            elif x not in map.keys(): # CLose bracker
                if(len(list_1)==0): return False
                if x != list_1[-1]:
                    return False
                list_1.pop()
        print(list_1)
        if len(list_1) > 0:
            return False 
        return True

    
        